block-producer/src/batch_builder/errors.rs

```diff
@@ -1,15 +1,17 @@
+use miden_objects::transaction::ProvenTransaction;
 use miden_vm::crypto::MerkleError;
 use thiserror::Error;
 
 use crate::MAX_NUM_CREATED_NOTES_PER_BATCH;
 
-#[derive(Error, Debug, PartialEq)]
+#[derive(Error, Debug)]
 pub enum BuildBatchError {
```

Comment:
Could we add a comment here explaining what the internal `Vec<ProvenTransaction>` represents?

Also, in the future, if the number of variants gets big, we may want to refactor this to look something like this:
```Rust
pub struct BuildBatchError {
    err_type: BuildBatchErrorType,
    proposed_txs: Vec<ProvenTransaction>,
}
```

Reply:
done